### ISSUE TITLE
Finish CashFusion integration

### ISSUE THREAD

**Issue by sneurlax** [cypherstack/stack_wallet#681]
CashFusion uses the [fusiondart](https://github.com/cypherstack/fusiondart) package, see [issues there](https://github.com/cypherstack/fusiondart/issues) for package-level TODOs, namely:

 - https://github.com/cypherstack/fusiondart/issues/15
 - https://github.com/cypherstack/fusiondart/issues/16
 - https://github.com/cypherstack/fusiondart/issues/17
 - https://github.com/cypherstack/fusiondart/issues/14
 - https://github.com/cypherstack/fusiondart/issues/2
 - ~~https://github.com/cypherstack/fusiondart/issues/18~~
 - https://github.com/cypherstack/fusiondart/issues/19
 - https://github.com/cypherstack/fusiondart/issues/20
 - ~~https://github.com/cypherstack/fusiondart/issues/21~~
 
Stack Wallet implementation TODOs:
 - [x] Polish UI state
   This can be checked once we can reliably loop without unhandled exceptions leading to lagged or stuck-incorrect uiState.
 - [x] Use UI inputs like CashFusion server, port, etc.
 - [x] Hook up cancel from UI
   Basically hooked up, but WIP because we don't wait for the cancel process to complete successfully.
 - [x] Parsing electrumx transactions to handle multiple inputs/outputs + special data (tokens, etc) + new transaction model in SW
 - [x] Ensure UTXOs passed to Fusion don't contain special data. Depends on the above
 - [x] Review TODOs in code
 
 Deployment TODOs:
  - [x] Merge fusiondart/staging to fusiondart/main
  - https://github.com/cypherstack/fusiondart/issues/3
  - [x] Replace fusiondart submodule with package as git ref

**Comment by sneurlax** [comment]
Now it seems like mostly bitcoincash wallets are left

**Comment by sneurlax** [comment]
Closed by https://github.com/cypherstack/stack_wallet/pull/685
